src/guide/security/generating-cryptographic-keys.md

```diff
@@ -0,0 +1,225 @@
+# Generating Cryptographic Keys
+
+In the realm of blockchain technology, cryptographic keys play a crucial role in upholding the security and authenticity of data transactions. With Iroha 2, you can create these vital keys for safeguarding your digital assets and communications.
+
+This section describes how to generate keys using `kagami`, a supporting tool shipped alongside Iroha 2.
+
+In the future, alternative methods of generating public keys shall be added.
+
+## Generating Cryptographic Keys with Kagami {#kagami}
+
+::: tip Note
+
+Since `kagami` does not come with a manual page, you can use the `--help` (`-h`) command to retrieve a brief summary of all the usable `kagami` parameters within the CLI you are using.
+
+:::
+
+To generate a new key pair having only just cloned the Iroha repo, run the following command from the project's `root` directory:
+
+```bash
+$ cargo run --bin kagami crypto
+```
+
+You can specify a number of different parameters to tailor the generated key pair to your specific needs. The following parameters are available:
+
+- `--algorithm` (`-a`): Specifies the algorithm used for the key pair generation and encryption. If no algorithm is specified, `ed25519` is used by default.
+    Can be one of the following:
+
+  - `ed25519` — <abbr title ="Elliptic Curve Cryptography">ECC</abbr> algorithm that utilizes the `Ed25519` curve, offering efficient and secure cryptographic operations for digital signatures and key exchange. If no algorithm is specified in a request, then `ed25519` is used by default.
+  > Learn more:\
+  [EdDSA > Ed25519 (Wikipedia)](https://en.wikipedia.org/wiki/EdDSA#Ed25519:~:text=.-,Ed25519,-%5Bedit%5D)\
+  [Ed25519: high-speed high-security signatures](https://ed25519.cr.yp.to/)
```

Comment:
- Indentation of `>` is not nested into the list item. And quotation block seems to not be really needed
- Using list instead of `\` line breaks

If you apply this, also update the below "learn more" items too.

```suggestion
    Learn more:
    - [EdDSA > Ed25519 (Wikipedia)](https://en.wikipedia.org/wiki/EdDSA#Ed25519:~:text=.-,Ed25519,-%5Bedit%5D)
    - [Ed25519: high-speed high-security signatures](https://ed25519.cr.yp.to/)
```

**Old:**

- ...
> Learn more:\
[EdDSA > Ed25519 (Wikipedia)](https://en.wikipedia.org/wiki/EdDSA#Ed25519:~:text=.-,Ed25519,-%5Bedit%5D)\
[Ed25519: high-speed high-security signatures](https://ed25519.cr.yp.to/)

**New:**

- ...
  Learn more:
  - [EdDSA > Ed25519 (Wikipedia)](https://en.wikipedia.org/wiki/EdDSA#Ed25519:~:text=.-,Ed25519,-%5Bedit%5D)
  - [Ed25519: high-speed high-security signatures](https://ed25519.cr.yp.to/)